examples/imagenet/train_imagenet.py

```diff
@@ -24,6 +25,78 @@
 import resnet50
 import resnext50
 
+try:
+    from nvidia import dali
+    from nvidia.dali import pipeline
+    from nvidia.dali import ops
+    _dali_available = True
+except ImportError:
+    _dali_available = False
+
+from chainer.backends import cuda
+import ctypes
+
+
+def _pair(x):
+    if hasattr(x, '__getitem__'):
+        return x
+    return x, x
+
+
+class ImagenetDaliPipeline(pipeline.Pipeline):
```

Comment:
nitpick: it will not work if `_dali_available == False` as `pipeline` is undefined.

Reply:
Thanks, I did not notice that. will add some workaround.